sdks/java/core/src/test/java/org/apache/beam/sdk/options/PipelineOptionsFactoryTest.java

```diff
@@ -752,10 +752,12 @@ public void testOutOfOrderArrays() {
   public static interface Lists extends PipelineOptions {
     List<String> getString();
     void setString(List<String> value);
+    List<Integer> getInteger();
+    void setInteger(List<Integer> value);
   }
 
   @Test
-  public void testList() {
+  public void testListString() {
     String[] args =
         new String[] {"--string=stringValue1", "--string=stringValue2", "--string=stringValue3"};
```

Comment:
Add tests for the singular valued list case for string and integer
Add tests for a rawtype collection (e.g. List)